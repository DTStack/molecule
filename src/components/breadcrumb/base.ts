import { prefixClaName, getBEMElement } from 'mo/common/className';

export const defaultBreadcrumbClassName = prefixClaName('breadcrumb');
export const breadcrumbItemClassName = getBEMElement(
    defaultBreadcrumbClassName,
    'item'
);
export const breadcrumbLabelClassName = getBEMElement(
    defaultBreadcrumbClassName,
    'label'
);
