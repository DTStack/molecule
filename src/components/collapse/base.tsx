import {
    prefixClaName,
    getBEMElement,
    getBEMModifier,
} from 'mo/common/className';

export const defaultCollapseClassName = prefixClaName('collapse');
export const collapseItemClassName = getBEMElement(
    defaultCollapseClassName,
    'item'
);

export const collapsePaneClassName = getBEMElement(
    defaultCollapseClassName,
    'pane'
);

export const collapsingClassName = getBEMModifier(
    collapsePaneClassName,
    'collapsing'
);

export const collapseActiveClassName = getBEMModifier(
    defaultCollapseClassName,
    'active'
);
export const collapseHeaderClassName = getBEMElement(
    defaultCollapseClassName,
    'header'
);
export const collapseTitleClassName = getBEMElement(
    collapseHeaderClassName,
    'title'
);
export const collapseContentClassName = getBEMElement(
    defaultCollapseClassName,
    'content'
);
export const collapseExtraClassName = getBEMElement(
    defaultCollapseClassName,
    'extra'
);
