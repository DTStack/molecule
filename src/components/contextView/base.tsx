import {
    prefixClaName,
    getBEMElement,
    getBEMModifier,
} from 'mo/common/className';

export const contextViewClass = prefixClaName('context-view');

export const contentClassName = getBEMElement(contextViewClass, 'content');

export const blockClassName = getBEMElement(contextViewClass, 'block');

export const shadowClassName = getBEMModifier(contextViewClass, 'shadow');
