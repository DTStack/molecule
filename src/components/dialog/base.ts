import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';
import { ConfrimType, ConfirmState } from './';

export const modalClassName = prefixClaName('modal');

export const confirmClassName = prefixClaName('confirm');

export const closeDialogClassName = getBEMElement(modalClassName, 'close');

export const closeDialogDescriptorClassName = getBEMModifier(
    closeDialogClassName,
    'x'
);
export const wrapDialogClassName = getBEMModifier(modalClassName, 'centered');

export const containerConfirmClassName = getBEMElement(
    confirmClassName,
    'container'
);

export const indicatorConfirmClassName = getBEMElement(
    confirmClassName,
    'indicator'
);

export const iconConfirmClassName = (type?: ConfrimType) =>
    typeof type === 'string'
        ? getBEMElement(confirmClassName, ConfirmState[type])
        : '';

export const contentConfirmClassName = getBEMElement(
    confirmClassName,
    'content'
);

export const messageConfirmClassName = getBEMElement(
    confirmClassName,
    'message'
);

export const btnsConfirmClassName = getBEMElement(confirmClassName, 'btns');

export const centeredConfirmClassName = getBEMModifier(
    confirmClassName,
    'centered'
);

export const textConfirmClassName = getBEMModifier(
    messageConfirmClassName,
    'text'
);

export const detailConfirmClassName = getBEMModifier(
    messageConfirmClassName,
    'detail'
);
