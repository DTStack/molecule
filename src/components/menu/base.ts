import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const defaultMenuClassName = prefixClaName('menu');
export const defaultSubMenuClassName = prefixClaName('sub-menu');

export const verticalMenuClassName = getBEMModifier(
    defaultMenuClassName,
    'vertical'
);
export const horizontalMenuClassName = getBEMModifier(
    defaultMenuClassName,
    'horizontal'
);

export const defaultMenuItemClassName = getBEMElement(
    defaultMenuClassName,
    'item'
);
export const defaultDividerClassName = getBEMElement(
    defaultMenuClassName,
    'divider'
);
export const checkClassName = getBEMElement(defaultMenuClassName, 'check');
export const disabledClassName = getBEMModifier(
    defaultMenuItemClassName,
    'disabled'
);
export const activeClassName = getBEMModifier(
    defaultMenuItemClassName,
    'active'
);
export const labelClassName = getBEMElement(defaultMenuClassName, 'label');
export const menuContentClassName = getBEMElement(
    defaultMenuClassName,
    'content'
);
export const keybindingClassName = getBEMElement(
    defaultMenuClassName,
    'keybinding'
);
export const indicatorClassName = getBEMElement(
    defaultMenuClassName,
    'indicator'
);
