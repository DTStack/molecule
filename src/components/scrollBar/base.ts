import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const baseClassName = prefixClaName('scrollBar');

export const scrollBarClassName = getBEMElement(baseClassName, 'wrapper');
export const scrollBarContainerClassName = getBEMElement(
    baseClassName,
    'container'
);
export const scrollBarShadowClassName = getBEMElement(baseClassName, 'shadow');
export const scrollBarTrackClassName = getBEMElement(baseClassName, 'track');
export const scrollBarThumbClassName = getBEMElement(baseClassName, 'thumb');

export const scrollBarContainerHorizontalClassName = getBEMModifier(
    scrollBarContainerClassName,
    'horizontal'
);

export const scrollBarContainerVerticalClassName = getBEMModifier(
    scrollBarContainerClassName,
    'vertical'
);

export const scrollBarTrackHiddenClassName = getBEMModifier(
    scrollBarTrackClassName,
    'hidden'
);

export const scrollBarTrackVerticalClassName = getBEMModifier(
    scrollBarTrackClassName,
    'vertical'
);

export const scrollBarTrackHorizontalClassName = getBEMModifier(
    scrollBarTrackClassName,
    'horizontal'
);

export const scrollBarShadowHiddenClassName = getBEMModifier(
    scrollBarShadowClassName,
    'hidden'
);
