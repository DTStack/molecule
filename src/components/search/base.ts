import { getBEMElement, prefixClaName } from 'mo/common/className';

export const defaultSearchClassName = prefixClaName('search');

export const baseInputClassName = getBEMElement(
    defaultSearchClassName,
    'input'
);

export const inputGroupClassName = getBEMElement(
    defaultSearchClassName,
    'group'
);

export const searchToolBarClassName = getBEMElement(
    defaultSearchClassName,
    'toolbar'
);

export const replaceContainerClassName = getBEMElement(
    defaultSearchClassName,
    'replace'
);

export const searchTargetContainerClassName = getBEMElement(
    defaultSearchClassName,
    'target'
);

export const replaceBtnClassName = getBEMElement(
    replaceContainerClassName,
    'button'
);

export const validationBaseInputClassName = getBEMElement(
    defaultSearchClassName,
    'base'
);

export const validationInfoInputClassName = getBEMElement(
    defaultSearchClassName,
    'info'
);

export const validationWarningInputClassName = getBEMElement(
    defaultSearchClassName,
    'warning'
);

export const validationErrorInputClassName = getBEMElement(
    defaultSearchClassName,
    'error'
);
