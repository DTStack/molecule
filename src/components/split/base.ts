import {
    getBEMElement,
    getBEMModifier,
    prefixClaName,
} from 'mo/common/className';

export const splitClassName = prefixClaName('split');

export const splitDraggingClassName = getBEMModifier(
    splitClassName,
    'dragging'
);
export const splitVerticalClassName = getBEMModifier(
    splitClassName,
    'vertical'
);
export const splitHorizontalClassName = getBEMModifier(
    splitClassName,
    'horizontal'
);

export const paneItemClassName = getBEMElement(splitClassName, 'pane');
export const sashItemClassName = getBEMElement(splitClassName, 'sash');

export const sashVerticalClassName = getBEMModifier(
    sashItemClassName,
    'vertical'
);
export const sashHorizontalClassName = getBEMModifier(
    sashItemClassName,
    'horizontal'
);
export const sashDisabledClassName = getBEMModifier(
    sashItemClassName,
    'disabled'
);
export const sashHoverClassName = getBEMModifier(sashItemClassName, 'hover');
