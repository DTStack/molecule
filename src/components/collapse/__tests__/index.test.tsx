import React from 'react';
import { act, cleanup, fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Collapse } from '../index';
import { sleep } from '@test/utils';
import { sashItemClassName } from 'mo/components/split/base';
import { collapsePaneClassName } from '../base';

afterEach(cleanup);

describe('Test The Collapse Component', () => {
    beforeAll(() => {
        // @ts-ignore
        window.innerHeight = 500;
    });

    let original;
    const observerFnCollection: any[] = [];
    beforeEach(() => {
        original = HTMLElement.prototype.getBoundingClientRect;
        // @ts-ignore
        HTMLElement.prototype.getBoundingClientRect = () => ({
            height: 500,
        });

        global.ResizeObserver = jest.fn().mockImplementation((fn) => {
            fn();
            observerFnCollection.push(fn);
            return {
                observe: jest.fn(),
                unobserve: jest.fn(),
                disconnect: jest.fn(),
            };
        });
    });

    afterEach(() => {
        HTMLElement.prototype.getBoundingClientRect = original;
        observerFnCollection.length = 0;
    });

    test('Match Snapshot', () => {
        const { asFragment } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    { id: 'mock2', name: 'test2' },
                ]}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('Match Snapshot Empty', () => {
        const { asFragment } = render(<Collapse />);

        expect(asFragment()).toMatchSnapshot();
    });

    test('Should uncollapsing the mock2', async () => {
        const mockFn = jest.fn();
        const mockResize = jest.fn();
        const { container } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    { id: 'mock3', name: 'test3' },
                ]}
                onCollapseChange={mockFn}
                onResize={mockResize}
            />
        );

        const collaspeItem = container.querySelector(
            'div[data-collapse-id="mock2"]'
        );

        expect(collaspeItem?.parentElement?.style.height).toBe('26px');

        await act(async () => {
            // uncollasing mock2
            fireEvent.click(collaspeItem?.childNodes[0]!);
            await sleep(300);
        });
        expect(mockFn).toBeCalled();
        expect(mockFn.mock.calls[0][0]).toEqual([
            undefined,
            'mock2',
            undefined,
        ]);

        expect(mockResize).toBeCalled();
        expect(mockResize.mock.calls[0][0]).toEqual([26, 448, 26]);

        mockFn.mockClear();
        mockResize.mockClear();

        await act(async () => {
            // collapsing mock2 to restore default status
            fireEvent.click(collaspeItem?.childNodes[0]!);
            await sleep(300);
        });
        expect(mockFn).toBeCalled();
        expect(mockFn.mock.calls[0][0]).toEqual([
            undefined,
            undefined,
            undefined,
        ]);

        expect(mockResize).toBeCalled();
        expect(mockResize.mock.calls[0][0]).toEqual([26, 26, 26]);

        // collapsing mock2 and mock3
        await act(async () => {
            fireEvent.click(collaspeItem?.childNodes[0]!);
            await sleep(300);
        });

        await act(async () => {
            const mock3 = container.querySelector(
                'div[data-collapse-id="mock3"]'
            );
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        expect(mockResize).toBeCalled();
        // divided the remaining space by grow number
        expect(mockResize.mock.calls[2][0]).toEqual([26, 316, 158]);
    });

    test('Should support to change size', async () => {
        const mockResize = jest.fn();
        const { container, getByTestId } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                        renderPanel: () => <div data-testid="test"></div>,
                    },
                ]}
                onResize={mockResize}
            />
        );
        expect(getByTestId('test')).toBeInTheDocument();

        await act(async () => {
            const mock2 = container.querySelector(
                'div[data-collapse-id="mock2"]'
            );
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        await act(async () => {
            const mock3 = container.querySelector(
                'div[data-collapse-id="mock3"]'
            );
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        mockResize.mockClear();
        const wrapper = container.querySelector(`.${collapsePaneClassName}`)!;
        const sashs = container.querySelectorAll(`.${sashItemClassName}`);
        fireEvent.mouseDown(sashs[1], { screenY: 0 });
        fireEvent.mouseMove(wrapper, { screenY: 10 });
        fireEvent.mouseUp(wrapper);

        expect(mockResize).toBeCalled();
        expect(mockResize.mock.calls[0][0]).toEqual([26, 326, 148]);
    });

    test('Should NOT trigger onChange', async () => {
        const mockResize = jest.fn();
        const { container, getByTestId } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                        renderPanel: () => <div data-testid="test"></div>,
                    },
                ]}
                onResize={mockResize}
            />
        );
        expect(getByTestId('test')).toBeInTheDocument();

        await act(async () => {
            const mock2 = container.querySelector(
                'div[data-collapse-id="mock2"]'
            );
            const mock3 = container.querySelector(
                'div[data-collapse-id="mock3"]'
            );
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        mockResize.mockClear();
        const wrapper = container.querySelector(`.${collapsePaneClassName}`)!;
        const sashs = container.querySelectorAll(`.${sashItemClassName}`);
        // ensure when there is no changes in sizes it won't trigger onResize event
        fireEvent.mouseDown(sashs[1]);
        fireEvent.mouseMove(wrapper, { screenX: 0, screenY: 0 });
        fireEvent.mouseUp(wrapper);
        expect(mockResize).not.toBeCalled();
    });

    test('Should resize grow:2 pane', async () => {
        const { container } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                    },
                ]}
            />
        );

        const mock2 = container.querySelector('div[data-collapse-id="mock2"]');
        await act(async () => {
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock2?.parentElement?.style.height).toBe('448px');

        await act(async () => {
            // @ts-ignore
            HTMLElement.prototype.getBoundingClientRect = () => ({
                height: 1000,
            });
            observerFnCollection.forEach((f) => f());
            await sleep(150);
        });

        expect(mock2?.parentElement?.style.height).toBe('948px');
    });

    test('Should NOT render hidden pane', async () => {
        const { container } = render(
            <Collapse
                data={[
                    {
                        id: 'mock1',
                        name: 'test1',
                        config: { grow: 0 },
                        hidden: true,
                    },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                    },
                ]}
            />
        );
        const mock1 = container.querySelector('div[data-collapse-id="mock1"]');
        expect(mock1?.parentElement?.style.height).toBe('0px');

        await act(async () => {
            const mock2 = container.querySelector(
                'div[data-collapse-id="mock2"]'
            );
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        // didn't effect the hidden pane
        expect(mock1?.parentElement?.style.height).toBe('0px');

        await act(async () => {
            const mock2 = container.querySelector(
                'div[data-collapse-id="mock2"]'
            );
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock1?.parentElement?.style.height).toBe('0px');
    });

    test('Should support to collapse a auto height pane', async () => {
        const { container } = render(
            <Collapse
                data={[
                    {
                        id: 'mock1',
                        name: 'test1',
                        config: { grow: 0 },
                        renderPanel: () => (
                            <div data-content="mock1" style={{ height: 500 }}>
                                1
                            </div>
                        ),
                    },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                    },
                ]}
            />
        );

        const mock1 = container.querySelector('div[data-collapse-id="mock1"]');
        expect(mock1?.parentElement?.style.height).toBe('26px');

        await act(async () => {
            fireEvent.click(mock1?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock1?.parentElement?.style.height).toBe('220px');

        await act(async () => {
            // collapsing it
            fireEvent.click(mock1?.childNodes[0]!);
            await sleep(300);
        });

        await act(async () => {
            // @ts-ignore
            HTMLElement.prototype.getBoundingClientRect = () => ({
                height: 100,
            });

            // re-uncollapse it
            fireEvent.click(mock1?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock1?.parentElement?.style.height).toBe('126px');
    });

    test('Should NOT uncollapse the empty grow-0 pane', async () => {
        const { container } = render(
            <Collapse
                data={[
                    {
                        id: 'mock1',
                        name: 'test1',
                        config: { grow: 0 },
                        renderPanel: () => null,
                    },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    {
                        id: 'mock3',
                        name: 'test3',
                    },
                ]}
            />
        );
        const mock1 = container.querySelector('div[data-collapse-id="mock1"]');
        expect(mock1?.parentElement?.style.height).toBe('26px');

        await act(async () => {
            fireEvent.click(mock1?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock1?.parentElement?.style.height).toBe('26px');
    });

    test('Should support to cache the adjusted size', async () => {
        const { container } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    { id: 'mock2', name: 'test2', config: { grow: 2 } },
                    { id: 'mock3', name: 'test3' },
                ]}
            />
        );

        // uncollaspe the mock2 and mock3
        const mock2 = container.querySelector('div[data-collapse-id="mock2"]');
        const mock3 = container.querySelector('div[data-collapse-id="mock3"]');
        await act(async () => {
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        await act(async () => {
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock2?.parentElement?.style.height).toBe('316px');
        expect(mock3?.parentElement?.style.height).toBe('158px');

        // adjust the sizes of the mock2 and mock3
        const wrapper = container.querySelector(`.${collapsePaneClassName}`)!;
        const sashs = container.querySelectorAll(`.${sashItemClassName}`);
        fireEvent.mouseDown(sashs[1], { screenY: 0 });
        fireEvent.mouseMove(wrapper, { screenY: 10 });
        fireEvent.mouseUp(wrapper);

        expect(mock2?.parentElement?.style.height).toBe('326px');
        expect(mock3?.parentElement?.style.height).toBe('148px');

        await act(async () => {
            // collapse the mock3
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock2?.parentElement?.style.height).toBe('448px');
        expect(mock3?.parentElement?.style.height).toBe('26px');

        await act(async () => {
            // uncollapse the mock3
            fireEvent.click(mock3?.childNodes[0]!);
            await sleep(300);
        });

        expect(mock2?.parentElement?.style.height).toBe('326px');
        expect(mock3?.parentElement?.style.height).toBe('148px');
    });

    test('Should support to onToolbarClick', async () => {
        const mockFn = jest.fn();
        const { container, getByTestId } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    {
                        id: 'mock2',
                        name: 'test2',
                        config: { grow: 2 },
                        toolbar: [
                            { id: 'toolbar1', 'data-testid': 'toolbar1' },
                        ],
                    },
                    { id: 'mock3', name: 'test3' },
                ]}
                onToolbarClick={mockFn}
            />
        );

        await act(async () => {
            // uncollaspe the mock2 and mock3
            const mock2 = container.querySelector(
                'div[data-collapse-id="mock2"]'
            );
            fireEvent.click(mock2?.childNodes[0]!);
            await sleep(300);
        });

        expect(getByTestId('toolbar1')).toBeInTheDocument();
        fireEvent.click(getByTestId('toolbar1'));
        expect(mockFn).toBeCalled();
        expect(mockFn.mock.calls[0][0]).toEqual(
            expect.objectContaining({
                id: 'toolbar1',
                'data-testid': 'toolbar1',
            })
        );
        expect(mockFn.mock.calls[0][1]).toEqual({
            id: 'mock2',
            name: 'test2',
            config: { grow: 2 },
            toolbar: [{ id: 'toolbar1', 'data-testid': 'toolbar1' }],
        });
    });

    test('Should support to set activePanelKeys', async () => {
        const { container } = render(
            <Collapse
                data={[
                    { id: 'mock1', name: 'test1', config: { grow: 0 } },
                    {
                        id: 'mock2',
                        name: 'mock2',
                    },
                    { id: 'mock3', name: 'test3' },
                ]}
                activePanelKeys={['mock2']}
            />
        );
        const metaData = (
            container.querySelector('div[data-collapse-id="mock2"]') as any
        )?.dataset.collapseId;

        expect(metaData).toBe('mock2');
    });
});
