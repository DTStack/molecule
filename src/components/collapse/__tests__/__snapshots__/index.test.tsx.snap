// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test The Collapse Component Match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="mo-collapse"
  >
    <div
      class="mo-split mo-split--horizontal"
    >
      <div
        class="mo-split__pane mo-collapse__pane"
        style="height: 26px; top: 0px;"
      >
        <div
          class="mo-collapse__item"
          data-collapse-id="mock1"
        >
          <div
            class="mo-collapse__header"
            tabindex="0"
          >
            <span
              class="codicon codicon-chevron-right"
            />
            <span
              class="mo-collapse__header__title"
            >
              test1
            </span>
            <div
              class="mo-collapse__extra"
            />
          </div>
          <div
            class="mo-collapse__content"
            data-collapse-index="0"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="mo-split__sash mo-split__sash--disabled mo-split__sash--horizontal"
        role="Resizer"
        style="height: 4px; top: 24px;"
      />
      <div
        class="mo-split__pane mo-collapse__pane"
        style="height: 26px; top: 26px;"
      >
        <div
          class="mo-collapse__item"
          data-collapse-id="mock2"
        >
          <div
            class="mo-collapse__header"
            tabindex="0"
          >
            <span
              class="codicon codicon-chevron-right"
            />
            <span
              class="mo-collapse__header__title"
            >
              test2
            </span>
            <div
              class="mo-collapse__extra"
            />
          </div>
          <div
            class="mo-collapse__content"
            data-collapse-index="1"
            tabindex="0"
          />
        </div>
      </div>
      <div
        class="mo-split__sash mo-split__sash--disabled mo-split__sash--horizontal"
        role="Resizer"
        style="height: 4px; top: 50px;"
      />
      <div
        class="mo-split__pane mo-collapse__pane"
        style="height: 26px; top: 52px;"
      >
        <div
          class="mo-collapse__item"
          data-collapse-id="mock2"
        >
          <div
            class="mo-collapse__header"
            tabindex="0"
          >
            <span
              class="codicon codicon-chevron-right"
            />
            <span
              class="mo-collapse__header__title"
            >
              test2
            </span>
            <div
              class="mo-collapse__extra"
            />
          </div>
          <div
            class="mo-collapse__content"
            data-collapse-index="2"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test The Collapse Component Match Snapshot Empty 1`] = `
<DocumentFragment>
  <div
    class="mo-collapse"
  >
    <div
      class="mo-split mo-split--horizontal"
    />
  </div>
</DocumentFragment>
`;
